### PR TITLE
znapzend: add service plist

### PR DESCRIPTION
**Pull request by claui**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR introduces a plist for the znapzend service so it can be used with `brew services`.

Note that the timeout for SSH connections has been set to 120 seconds, a value I have tuned during a few months of real-world testing on several Macs. (In particular, delays in excess of one minute have been observed in a Time-Machine-like scenario, e. g. the receiving zpool is on another Mac on the local network, which may need time to wake up first before serving the SSH connection.)

### CODE REVIEW

File: Formula/znapzend.rb [hunk 1]
```diff
@@ -20,6 +21,49 @@ def install
     system "make", "install"
   end
 
+  def post_install
+    (var/"log/#{name}").mkpath
```
Comment (ilovezfs) [inline Formula/znapzend.rb]: We generally do not want to use `name` like this in formulae and just hard code it.

File: Formula/znapzend.rb [hunk 2]
```diff
@@ -20,6 +21,49 @@ def install
...
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN"
+    "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>EnvironmentVariables</key>
+        <dict>
+          <key>PATH</key>
+          <string>/usr/bin:/bin:/usr/sbin:/sbin:#{HOMEBREW_PREFIX}/bin</string>
```
Comment (ilovezfs) [inline Formula/znapzend.rb]: does this still work when HOMEBREW_PREFIX is not /usr/local? i.e. does znapzend know where to find `zpool` if it's not in PATH here?

File: Formula/znapzend.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@ class Znapzend < Formula
   homepage "https://www.znapzend.org/"
   url "https://github.com/oetiker/znapzend/releases/download/v0.19.1/znapzend-0.19.1.tar.gz"
   sha256 "93e3ec3c6f5cdf6973f72a6b764c49dc6545f2a0a2e0267a1382d471b930efea"
+  revision 1
```
Comment (ilovezfs) [inline Formula/znapzend.rb]: revision isn't needed because the bottles aren't affected

File: Formula/znapzend.rb [hunk 4]
```diff
@@ -20,6 +21,49 @@ def install
     system "make", "install"
   end
 
+  def post_install
+    (var/"log/#{name}").mkpath
+    (var/"run/#{name}").mkpath
+  end
+
+  plist_options(:startup => true, :manual => "znapzend --daemonize")
```
Comment (ilovezfs) [inline Formula/znapzend.rb]: the parentheses are not idiomatic and I assume znapzend needs sudo to start manually

File: Formula/znapzend.rb [hunk 5]
```diff
@@ -20,6 +21,49 @@ def install
...
+        <string>#{var}/log/#{name}/#{name}.err.log</string>
+        <key>StandardOutPath</key>
+        <string>#{var}/log/#{name}/#{name}.out.log</string>
+        <key>ThrottleInterval</key>
+        <integer>30</integer>
+        <key>WorkingDirectory</key>
+        <string>#{var}/run/#{name}</string>
+      </dict>
+    </plist>
+    EOS
```
Comment (ilovezfs) [inline Formula/znapzend.rb]: off by two spaces

File: Formula/znapzend.rb (hunk 3 above)
Comment (claui) [reply Formula/znapzend.rb]: Will `brew upgrade` still pick up the new formula if we remove `revision 1`?

Comment (ilovezfs) [reply Formula/znapzend.rb]: No but I don't think just adding a plist warrants an upgrade. The changes will kick in for new installs or on `brew reinstall`.

File: Formula/znapzend.rb (hunk 2 above)
Comment (claui) [reply Formula/znapzend.rb]: Good point. It doesn’t unless `zpool` happens to live in `HOMEBREW_PREFIX`, too, which is unlikely; I only just realized the `openzfs` cask has `/usr/local/bin` hard-coded.

I’ll insert `/usr/local/bin`.

File: Formula/znapzend.rb (hunk 3 above)
Comment (claui) [reply Formula/znapzend.rb]: Fair enough.

File: Formula/znapzend.rb (hunk 4 above)
Comment (claui) [reply Formula/znapzend.rb]: > the parentheses are not idiomatic

I don’t really see an alternative; neither could I find any in the core tap:

```
grep -R 'var.*log.*mkpath' "$(brew --repository)/Library/Taps/homebrew/homebrew-core"
```

> assume znapzend needs sudo to start manually

In most cases, yes. Good point.

Comment (ilovezfs) [reply Formula/znapzend.rb]: you mean like
```
stubby.rb:  plist_options :startup => true, :manual => "sudo stubby -C #{HOMEBREW_PREFIX}/etc/stubby/stubby.yml"
```

File: Formula/znapzend.rb (hunk 1 above)
Comment (claui) [reply Formula/znapzend.rb]: I have no strong opinion on that.  
Just for clarification, would you say the plist should be changed in the same way?

Comment (ilovezfs) [reply Formula/znapzend.rb]: yes

File: Formula/znapzend.rb [hunk 6]
```diff
@@ -20,6 +20,49 @@ def install
     system "make", "install"
   end
 
+  def post_install
+    (var/"log/znapzend").mkpath
+    (var/"run/znapzend").mkpath
+  end
+
+  plist_options(:startup => true, :manual => "sudo znapzend --daemonize")
```
Comment (ilovezfs) [inline Formula/znapzend.rb]: parentheses should still be removed here

**Comment by ilovezfs** [comment]
Thanks again @claui!

**Pull request merged**